src/empirical.jl

```diff
@@ -1,68 +1,142 @@
 # Empirical estimation of CDF and PDF
 
-## Empirical CDF
+"""
+Empirical Cumulative Distribution Function (ECDF).
+
+Represents ECDF constructed from random variable samples by [`ecdf`](@ref).
+
+# Type parameters
+* `T`: the type of random variable values
+* `W`: the type of sample weights and CDF values
+* `I`: boolean indicating whether to interpolate
+  the CDF between neighboring samples (continuous distribution)
+  or not (discrete distribution)
+"""
+struct ECDF{T <: Real, W <: Real, I}
+    # sorted random variable values and associated probabilities.
+    # The tuple provides:
+    #  - `x[i]` (value of random variable)
+    #  - `ECDF(x[i])`
+    #  - `1/(x[i+1] - x[i])`
+    #  - `ECDF(x[i+1]) - ECDF(x[i])` (the weight of `x[i+1]`)
+    sorted_values::Vector{Tuple{T, W, W, W}}
+end
+
+# type of value weights and CDF values
+weighttype(::Type{<:ECDF{<:Any, W}}) where W = W
+weighttype(ecdf::ECDF) = weighttype(typeof(ecdf))
 
-struct ECDF{T <: AbstractVector{<:Real}, W <: AbstractWeights{<:Real}}
-    sorted_values::T
-    weights::W
+# whether to interpolate between the neighboring ECDF values
+isinterpolating(::Type{<:ECDF{<:Any, <:Any, I}}) where I = I
+isinterpolating(ecdf::ECDF) = isinterpolating(typeof(ecdf))
+
+function Base.show(io::IO, e::ECDF)
+    print(io, typeof(e))
+    print(io, "(")
+    print(io, length(e.sorted_values))
+    println(io, " values)")
```

Comment:
```suggestion
    print(io, "(")
    print(io, length(e.sorted_values))
    println(io, " values)")
```
To avoid making it look as if values was and argument to the constructor:
```suggestion
    print(io, " with ")
    print(io, length(e.sorted_values))
    println(io, " values")
```

Also maybe "points" is better than "values"? I.e. it's not the number of values in the input but the number of unique values, right?